src/main/java/org/cyclops/cyclopscore/tileentity/CyclopsTileEntity.java

```diff
@@ -287,18 +292,19 @@ protected EnumFacing transformFacingForRotation(EnumFacing facing) {
     }
 
     @Override
-    public boolean hasCapability(Capability<?> capability, EnumFacing facing) {
-        return (capabilities != null && capabilities.contains(capability, transformFacingForRotation(facing)))
-                || (facing != null && capabilities != null && capabilities.contains(capability, null))
+    public boolean hasCapability(@Nonnull Capability<?> capability, EnumFacing facing) {
+        return (sidedCapabilities != null && sidedCapabilities.contains(capability, transformFacingForRotation(facing)))
```

Comment:
Only query `sidedCapabilities` if `facing` is not null, otherwise, query `capabilities`.